network-api/networkapi/wagtailpages/pagemodels/libraries/rcc/detail_page.py

```diff
@@ -115,64 +107,144 @@ class RCCDetailPage(BasePage):
-        #             [index.SearchField("name")],
-        #         )
-        #     ],
-        # ),
-        # index.RelatedFields(
-        #     "related_topics",
-        #     [
-        #         index.RelatedFields(
-        #             "rcc_topic",
-        #             [index.SearchField("name")],
-        #         )
-        #     ],
-        # ),
+        index.RelatedFields(
+            "rcc_authors",
+            [
+                index.RelatedFields(
+                    "author_profile",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_content_types",
+            [
+                index.RelatedFields(
+                    "content_type",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_curricular_areas",
+            [
+                index.RelatedFields(
+                    "curricular_area",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_topics",
+            [
+                index.RelatedFields(
+                    "rcc_topic",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
     ]
 
-    # def get_context(self, request):
-    #     context = super().get_context(request)
-    #     context["authors_index"] = authors_index.RCCAuthorsIndexPage.objects.first()
-    #     context["rcc_authors"] = self.get_rcc_authors()
-    #     context["rcc_author_names"] = self.get_rcc_author_names()
-    #     context["content_type_names"] = self.get_related_content_types_names()
-    #     return context
+    def get_context(self, request):
+        context = super().get_context(request)
+        context["authors_index"] = authors_index.RCCAuthorsIndexPage.objects.first()
+        context["rcc_authors"] = self.get_rcc_authors()
+        context["rcc_author_names"] = self.get_rcc_author_names()
+        context["content_type_names"] = self.get_related_content_types_names()
+        return context
 
-    # def get_rcc_authors(self):
-    #     rcc_author_profiles = localize_queryset(
-    #         Profile.objects.prefetch_related("authored_rcc_entries").filter(authored_rcc_entries__rcc_detail_page=self)
-    #     )
-    #     return rcc_author_profiles
+    def get_rcc_authors(self):
+        rcc_author_profiles = localize_queryset(
+            Profile.objects.prefetch_related("authored_rcc_articles").filter(
+                authored_rcc_articles__rcc_detail_page=self
+            )
+        )
+        return rcc_author_profiles
 
-    # def get_rcc_author_names(self):
-    #     return [ra.author_profile.name for ra in self.rcc_authors.all()]
+    def get_rcc_author_names(self):
+        return [ra.author_profile.name for ra in self.rcc_authors.all()]
```

Comment:
Have you considered using the method above to retrieve the author objects? It seems odd that there are two ways how they are retrieved. If we need the values multiple times it might be work caching the result too.

Reply:
This will be handled in the common patterns refactoring

Reply:
Sorry, I am actually not sure I understand. 

I meant this. 
```suggestion
        return [ra.author_profile.name for ra in self.get_rcc_authors()]
```

What would be the common pattern refactoring here?